app/src/main/java/com/android/example/FakeContentProviderFilenameSanitization.kt

```diff
@@ -0,0 +1,68 @@
+package com.android.example
+
+import android.database.Cursor
+import android.provider.OpenableColumns
+import java.io.File
+
+
+class FakeContentProviderFilenameSanitization {
+    private fun trustProvidedFilename_shouldWarn(cursor: Cursor) {
+        val id = cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME)
+        val fileName = cursor.getString(id)
+
+        val fileObject = File("./", fileName)
+    }
+
+    private fun cursorFilenameAssignedToVariables(cursor: Cursor) {
+        val fileName = cursor.getString(cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME))
+
+        val fileObject = File("./", fileName)
+    }
+
+    private fun additionalPerformedOperations(cursor: Cursor) {
+        var temp = 1 + 2
+
+        val id = cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME)
+        println(temp)
+        val fileName = cursor.getString(id)
+
+        temp += 2
+        val fileObject = File("./", fileName)
+    }
+
+    private fun callChainStartingAtContext(cursor: Cursor) {
+        val id = cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME)
+        val fileName = cursor.getString(id)
+
+        val fileObject = File("./", fileName)
+    }
+
+    private fun unsanitizedFilenameUsedTwice(cursor: Cursor) {
+        val id = cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME)
+        val fileName = cursor.getString(id)
+
+        val fileObject = File("./", fileName)
+        File("./", fileName)
+    }
+
+    private fun columnNameAsString(cursor: Cursor) {
+        val id = cursor.getColumnIndex("_display_name")
+        val fileName = cursor.getString(id)
+
+        val fileObject = File("./", fileName)
+    }
+
+    private fun usingProvidedFilenameAsQualifiedExpression(cursor: Cursor) {
+        val id = cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME)
+
+        val fileObject = File("./", cursor.getString(id))
+    }
+
+    private fun useGetColumnIndexOrThrow(cursor: Cursor) {
+        val id = cursor.getColumnIndexOrThrow(OpenableColumns.DISPLAY_NAME)
+        val fileName = cursor.getString(id)
+
+        val fileObject = File("./", fileName)
+    }
+
+}
```

Comment:
Nit: Can you add a new line at the end of the file please?

Reply:
Done!

Reply:
Hmm, I added a newline and it shows in Android Studio but it doesn't appear to have reflected it, so I've added two newlines now just to have the trailing newline show up properly :/